packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts

```diff
@@ -41,10 +42,10 @@ export abstract class LoadBalancer {
    *
    * @param albTargetGroup an ALB Target Group
    */
-  public static application(albTargetGroup: elbv2.ApplicationTargetGroup): LoadBalancer {
+  public static application(albTargetGroup: elbv2.IApplicationTargetGroup): LoadBalancer {
     class AlbLoadBalancer extends LoadBalancer {
       public readonly generation = LoadBalancerGeneration.SECOND;
-      public readonly name = albTargetGroup.targetGroupName;
+      public readonly name = targetGruopNameFromArn(albTargetGroup.targetGroupArn);
```

Comment:
I don't particularly like this solution. I actually think this is putting the logic of how to get the name of the Target Group from its ARN in the wrong place.

Let's do this a little bit differently:

1. Let's add a `targetGroupName` property to `ITargetGroup`.
2. In imported Target Groups, let's add the logic of parsing it from the ARN of the Target Group.
3. Let's keep using `public readonly name = albTargetGroup.targetGroupName;` here, but now, this will Just Work.

Reply:
OK, I agree your proposal.👌
I will update soon with this approach.